libr/lang/p/pipe.c

```diff
@@ -36,21 +36,25 @@ static HANDLE myCreateChildProcess(const char * szCmdline) {
 static volatile BOOL bStopPipeLoop = FALSE;
 static HANDLE hPipeInOut = NULL;
 static HANDLE hproc = NULL;
+static HANDLE hConnected = NULL;
 #define PIPE_BUF_SIZE 8192
 
 static DWORD WINAPI WaitForProcThread(LPVOID lParam) {
-	WaitForSingleObject (hproc, INFINITE);
+	HANDLE hEvents[] = { hConnected, hproc };
+	DWORD dwEvent = WaitForMultipleObjects (R_ARRAY_SIZE (hEvents), hEvents, FALSE, INFINITE);
 
-	// Just in case the #!pipe target didn't actually connect to the pipe,
-	// we connect to the pipe here to satisfy the ConnectNamedPipe().
-	LPTSTR pipeName = calloc (128, sizeof (TCHAR));
-	GetEnvironmentVariable (TEXT ("R2PIPE_PATH"), pipeName, 128);
-	HANDLE pipe = CreateFile (pipeName, GENERIC_READ | GENERIC_WRITE, 0, NULL, OPEN_EXISTING, 0, NULL);
-	if (pipe != INVALID_HANDLE_VALUE) {
-		CloseHandle (pipe);
-	}
+	if (dwEvent == WAIT_OBJECT_0 + 1) {
+		// Just in case the #!pipe target didn't actually connect to the pipe,
+		// we connect to the pipe here to satisfy the ConnectNamedPipe().
+		LPTSTR pipeName = calloc (128, sizeof (TCHAR));
```

Comment:
leak of `pipeName`

Reply:
ok 311bbad7339d64f47d553c386c320d007b4842aa ... the entire function is slated for removal though